drivers/spi/spi_sedi.c

```diff
@@ -0,0 +1,435 @@
+	spi_context_release(&spi->ctx, ret);
+	pm_device_busy_clear(dev);
+
+	return ret;
+}
+
+static int spi_sedi_transceive(const struct device *dev,
+			       const struct spi_config *config,
+			       const struct spi_buf_set *tx_bufs,
+			       const struct spi_buf_set *rx_bufs)
+{
+	return transceive(dev, config, tx_bufs, rx_bufs, false, NULL, NULL);
+}
+
+#ifdef CONFIG_SPI_ASYNC
+static int spi_sedi_transceive_async(const struct device *dev,
+				     const struct spi_config *config,
+				     const struct spi_buf_set *tx_bufs,
+				     const struct spi_buf_set *rx_bufs,
+				     spi_callback_t cb,
+				     void *userdata)
+{
+	return transceive(dev, config, tx_bufs, rx_bufs, true, cb, userdata);
+}
+#endif /* CONFIG_SPI_ASYNC */
+
+static int spi_sedi_release(const struct device *dev,
+			    const struct spi_config *config)
+{
+	struct spi_sedi_data *spi = dev->data;
+
+	if (!spi_context_configured(&spi->ctx, config)) {
+		return -EINVAL;
+	}
+
+	/* Lock status to false */
+	spi->is_locked = false;
+	spi_context_unlock_unconditionally(&spi->ctx);
+
+	return 0;
+}
+
+extern void spi_isr(sedi_spi_t device);
+
+void spi_sedi_callback(uint32_t event, void *param)
+{
+	const struct device *dev = (const struct device *)param;
+	const struct spi_sedi_config *info = dev->config;
+	struct spi_sedi_data *spi = dev->data;
+	struct spi_context *ctx = &spi->ctx;
+	int error;
+
+	/* Update the semaphore of spi context */
+	if (event == SEDI_SPI_EVENT_DATA_LOST) {
+		error = -EIO;
+	} else {
+		error = 0;
+	}
+
+	if ((event == SEDI_SPI_EVENT_COMPLETE) ||
+	    (event == SEDI_SPI_EVENT_DATA_LOST)) {
+		spi_context_cs_control(&spi->ctx, true);
+		spi_context_complete(&spi->ctx, dev, error);
+	} else if (event == SEDI_SPI_EVENT_TX_FINISHED) {
+		spi_context_update_tx(ctx, 1, ctx->tx_len);
+		if (ctx->tx_len != 0) {
+			sedi_spi_update_tx_buf(info->spi_device, ctx->tx_buf,
+					       ctx->tx_len);
+			if ((ctx->rx_len == 0) &&
+			    (spi->rx_data_updated == false)) {
+				/* Update rx length if always no rx */
+				sedi_spi_update_rx_buf(info->spi_device, NULL,
+						       spi->rx_dummy_len);
+				spi->rx_data_updated = true;
+			}
+		} else if (spi->tx_data_updated == false) {
+			sedi_spi_update_tx_buf(info->spi_device, NULL,
+					       spi->tx_dummy_len);
+			spi->tx_data_updated = true;
+		}
+	} else if (event == SEDI_SPI_EVENT_RX_FINISHED) {
+		spi_context_update_rx(ctx, 1, ctx->rx_len);
+		if (ctx->rx_len != 0) {
+			sedi_spi_update_rx_buf(info->spi_device, ctx->rx_buf,
+					       ctx->rx_len);
+		}
+	}
+}
+
+static const struct spi_driver_api sedi_spi_api = {
+	.transceive = spi_sedi_transceive,
+#ifdef CONFIG_SPI_ASYNC
+	.transceive_async = spi_sedi_transceive_async,
+#endif  /* CONFIG_SPI_ASYNC */
+	.release = spi_sedi_release,
+};
+
+static int spi_sedi_init(const struct device *dev)
+{
+	const struct spi_sedi_config *info = dev->config;
+	struct spi_sedi_data *spi = dev->data;
+	int ret;
+
+	DEVICE_MMIO_MAP(dev, K_MEM_CACHE_NONE);
```

Comment:
to make this work with paging enabled, need to add DEVICE_MMIO_RAM and DEVICE_MMIO_ROM to spi_sedi_config and spi_sedi_data, see sedi_i2c.c.

Reply:
Will do.